sale_delivery_block/models/sale_order.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    @api.constrains('delivery_block_id')
+    def _check_not_auto_done(self):
+        for so in self:
+            if so.delivery_block_id and self.env['ir.values'].get_default(
```

Comment:
get_default can be done before the loop to avoid calling it several times.

Maybe, you can change the loop by an 'any' function.